gio/src/dbus_connection.rs

```diff
@@ -117,8 +117,20 @@ pub struct ActionGroupExportId(NonZeroU32);
 pub struct MenuModelExportId(NonZeroU32);
 #[derive(Debug, Eq, PartialEq)]
 pub struct FilterId(NonZeroU32);
+
 #[derive(Debug, Eq, PartialEq)]
-pub struct SignalSubscriptionId(NonZeroU32);
+struct SignalSubscriptionId(NonZeroU32);
+
+#[derive(Debug)]
+pub struct SignalSubscription(DBusConnection, Option<SignalSubscriptionId>);
```

Comment:
@sdroege In this implementation, the signal subscription keeps a strong reference to the connection so a signal subscription in and by itself is sufficient to keep the connection alive, in my understanding.

I think that's what you'd normally want to have, but I also thought about making the connection a weakref here.  Thoughts?

Reply:
No that part is clear. I mean, how would application code making use of all this actually look like? Keep in mind that I only have knowledge about how DBus works, I never really wrote code for it :)

Reply:
Ah I see. I'll make an example then which shows how to use this API in different ways 🙂